src/commands/birthday/_common.ts

```diff
@@ -0,0 +1,140 @@
+import { Cursor, Collection as MongoCollection } from 'mongodb';
+import ExecutionContext from '../../ExecutionContext';
+import * as Discord from '../../Discord';
+import { Collection } from '../../database';
+
+export type Birthday = {
+  _id: number;
+  users: string[];
+};
+
+/**
+ * Calculate the day of year from a date.
+ *
+ * @param date The date to convert.
+ */
+export const calculateDayOfYear = (date: string): number => {
```

Comment:
With all of these date manipulations, we could consider adding in a library like [Luxon](https://moment.github.io/luxon/) to simplify them, especially given that some of these are pretty complicated (and might be more complicated in the future).

Reply:
I introduced the [dayjs](https://github.com/iamkun/dayjs/) dependency in another PR. We could use it here if we need to